File: vash/protobuf/protoHandler.py
import data_pb2 as Schema

api1 = Schema.Entity()
api1.id = 1
api1.entityId = "1-2-3"
api1.name = "Spotify"
api1.entityType = "api"

collection1 = Schema.Entity()
collection1.id = 2
collection1.entityId = "1-2-4"
collection1.name = "Twitter"
collection1.entityType = "collection"

entities = Schema.Entities()

entities.entities.extend([api1, collection1])

with open("./data_binary", "rb") as fd:
    fileContents = str(fd.read, 'ascii')
    x = Schema.Entities().ParseFromString(fileContents)

    print(x)

File: vash/protobuf/protoHandler.js
const Schema = require('./data_pb'),
    fs = require('fs');

try {
    const api1 = new Schema.Entity();

    api1.setId(1);
    api1.setName("Spotify");
    api1.setEntitytype("api");
    api1.setEntityid("1-2-3");
    
    const collection1 = new Schema.Entity();
    
    collection1.setId(2);
    collection1.setName("Twitter");
    collection1.setEntitytype("collection");
    collection1.setEntityid("1-2-4");
    
    const entities = new Schema.Entities();
    
    entities.addEntities(api1);
    entities.addEntities(collection1);
    
    const bytes = entities.serializeBinary();
    
    fs.writeFileSync("data_binary.bin", bytes);
    
    const deserializedEntities = Schema.Entities.deserializeBinary(bytes);
    
    console.log(deserializedEntities.getEntitiesList());
}
catch (err) {
    console.log(err);
}
